packages/utils/src/event-bus/index.ts

```diff
@@ -57,6 +48,33 @@ export abstract class AbstractEventBusModuleService
       ...existingSubscribers,
       newSubscriberDescriptor,
     ])
+  }
+
+  public retrieveSubscribers(event: string | symbol) {
+    return this.eventToSubscribersMap_.get(event)
+  }
+
+  public subscribe(
+    eventName: string | symbol,
+    subscriber: EventBusTypes.Subscriber,
+    context?: EventBusTypes.SubscriberContext
+  ): this {
+    if (typeof subscriber !== `function`) {
+      throw new Error("Subscriber must be a function")
+    }
+    /**
+     * If context is provided, we use the subscriberId from it
+     * otherwise we generate a random using a ulid
+     */
+
+    const randId = ulid()
+    const event = eventName.toString()
+
+    this.storeSubscribers({
+      event,
+      subscriberId: context?.subscriberId ?? randId,
```

Comment:
suggestion(non-blocking): 
```suggestion
      subscriberId: context?.subscriberId ?? `${event}-${randId}`,
```